docs/sources/configuration/_index.md

```diff
@@ -911,13 +911,17 @@ wal:
   # CLI flag: -ingester.recover-from-wal
   [recover: <boolean> | default = false]
 
-# When WAL is enabled, should chunks be flushed to long-term storage on shutdown.
-# CLI flag: -ingester.flush-on-shutdown
-[flush_on_shutdown: <boolean> | default = false]
+  # When WAL is enabled, should chunks be flushed to long-term storage on shutdown.
+  # CLI flag: -ingester.flush-on-shutdown
+  [flush_on_shutdown: <boolean> | default = false]
 
-# Interval at which checkpoints should be created.
-# CLI flag: ingester.checkpoint-duration
-[checkpoint_duration: <duration> | default = 5m]
+  # Interval at which checkpoints should be created.
+  # CLI flag: ingester.checkpoint-duration
+  [checkpoint_duration: <duration> | default = 5m]
+
+  # Maximum memory size the WAL may use during replay. After hitting this it will flush data to storage before continuing.
```

Comment:
small nit: I think we should specify what memory is here. I believe this is compressed block and head content ? In theory more memory could be in used until runtime.GC

Reply:
Sure. We count bytes based on the log lines from the WAL segments that are not discarded after calls to `stream.Push` as well as calling the `Size()` method on the checkpointed chunks. I had the line `We suggest setting this to a high percentage (~75%) of available memory.` in the operations doc, but can include it here too. WDYT?

Reply:
LGTM